src/cumin/core.clj

```diff
@@ -13,16 +13,21 @@
 (defn ^:no-doc include* [ent mapping body-fn rows]
   (let [[pk fk] (first (dissoc mapping :as))
         pks (distinct (remove nil? (map pk rows)))
-        sub-q (-> (select* ent) (body-fn))]
+        sub-q (-> (select* ent) (body-fn))
+        results (when (seq pks)
+                  (-> sub-q
+                      (cond-> (not-any? #{:* :korma.core/* fk} (:fields sub-q))
+                        (update-in [:fields] conj fk))
+                      (where {fk [in pks]})
+                      (select)))
+        [transformed-fk] (-> {fk nil}
```

Comment:
Nice catch! I'm worried a transformation may expect some data to be present and throw an exception when mimicking a record with `{fk nil}`.

How do you feel about removing the transforms, stitching the records together, and then applying the transformation to the stitched result set?

```clojure
(defn- ^:no-doc apply-transforms
  [results f k]
  (mapv (fn [row]
          (if (contains? row k)
            (update-in row [k] #(mapv f %))
            row))
        results))

(defn ^:no-doc include* [ent mapping body-fn rows]
  (let [[pk fk] (first (dissoc mapping :as))
        as-k (get mapping :as (keyword (:table ent)))
        pks (distinct (remove nil? (map pk rows)))
        sub-q (-> ent (dissoc :transforms) (select*) (body-fn))
        results (when (seq pks)
                  (-> sub-q
                      (cond-> (not-any? #{:* :korma.core/* fk} (:fields sub-q))
                        (update-in [:fields] conj fk))
                      (where {fk [in pks]})
                      (select)))]
    (if (seq results)
      (-> rows
          (stitch results as-k [pk fk])
          (apply-transforms (apply comp (:transforms ent)) as-k))
      rows)))
```